azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/DeployMojo.java

```diff
@@ -40,17 +46,26 @@ protected void doExecute() throws MojoExecutionException, MojoFailureException {
         // Prepare telemetries
         traceTelemetry(springAppClient, configuration);
         // Create or update new App
-        AppResourceInner app = springAppClient.createOrUpdateApp(configuration);
+        springAppClient.createOrUpdateApp(configuration);
         // Upload artifact
         final File toDeploy = Utils.getArtifactFromConfiguration(configuration);
         final ResourceUploadDefinitionInner uploadDefinition = springAppClient.uploadArtifact(toDeploy);
         // Create or update deployment
-        final SpringDeploymentClient deploymentClient = springAppClient.getActiveDeploymentClient();
-        final DeploymentResourceInner deployment = deploymentClient.createOrUpdateDeployment(configuration.getDeployment(), uploadDefinition);
-        // Update the app with new deployment
-        app = springAppClient.updateActiveDeployment(deployment.name());
+        final Deployment deploymentConfiguration = configuration.getDeployment();
+        final SpringDeploymentClient deploymentClient = springAppClient.getDeploymentClient(deploymentConfiguration.getDeploymentName());
+        if (deploymentClient.isDeploymentExist() || createInactive) {
+            // Update existing deployment or create an inactive one
+            deploymentClient.createOrUpdateDeployment(deploymentConfiguration, uploadDefinition);
+        } else if (springAppClient.getDeployments().size() == 0) {
+            // Create new deployment and active it
+            final DeploymentResourceInner deployment = deploymentClient.createOrUpdateDeployment(deploymentConfiguration, uploadDefinition);
+            springAppClient.updateActiveDeployment(deployment.name());
+        } else {
+            throw new IllegalArgumentException(DEPLOYMENT_NOT_EXIST);
+        }
+
         // Update deployment, show url
-        getLog().info(app.properties().url());
+        getLog().info(springAppClient.getApplicationUrl());
         getLog().info(deploymentClient.getDeploymentStatus().toString());
```

Comment:
the status should be shown in readable statement.

Reply:
This will be done in later PR with pending